java/lang/weak_hash_md5.yml

```diff
@@ -0,0 +1,50 @@
+imports:
+  - java_shared_lang_message_digest_md5_init
+patterns:
+  - pattern: |
+      $<MESSAGE_DIGEST_MD5_INIT>.update($<DATA_TYPE>)
+    filters:
+      - variable: MESSAGE_DIGEST_MD5_INIT
+        detection: java_shared_lang_message_digest_md5_init
+      - variable: DATA_TYPE
+        detection: datatype
+        scope: result
+  - pattern: |
+      $<MESSAGE_DIGEST_MD5_INIT>.update()
+    filters:
+      - variable: MESSAGE_DIGEST_MD5_INIT
+        detection: java_shared_lang_message_digest_md5_init
```

Comment:
A work-around for now because I couldn't get the `either` filter to work here - will confirm if this is a wider bug with Java (none of the other Java rules use `either`)

Reply:
That part should be language agnostic, weird.

Reply:
Pretty weird, yeah. I'll open an issue for this on Bearer repo 🤔